web/src/compose_recipient.js

```diff
@@ -108,18 +108,28 @@ export function update_on_recipient_change() {
     update_narrow_to_recipient_visibility();
 }
 
-export function check_stream_posting_policy_for_compose_box(stream_name) {
-    const stream = stream_data.get_sub_by_name(stream_name);
-    if (!stream) {
-        return;
+export function check_posting_policy_for_compose_box(stream_name) {
+    let can_post_messages;
+    // If stream_name is not passed, than it should be from a direct message.
+    if (stream_name === undefined) {
+        can_post_messages =
+            page_params.realm_private_message_policy !==
+            settings_config.private_message_policy_values.disabled.code;
+    } else {
+        const stream = stream_data.get_sub_by_name(stream_name);
+        if (!stream) {
+            return;
+        }
+        can_post_messages = stream_data.can_post_messages_in_stream(stream);
     }
-    const can_post_messages_in_stream = stream_data.can_post_messages_in_stream(stream);
-    if (!can_post_messages_in_stream) {
+    if (!can_post_messages) {
+        let msg = $t({defaultMessage: "You do not have permission to post in this stream."});
+        if (stream_name === undefined) {
+            msg = $t({defaultMessage: "Direct messages are disabled in this organization."});
```

Comment:
Let's call this variable `banner_text`; `msg` can easily be confused with a Zulip message object.

I think it would also be better to instead have the blocks that set `can_post_messages` set these strings as well if appropriate.

Finally, it might make sense to add a bonus commit to reorder this function to use the early return pattern -- i.e. make this conditional be `if (can_post_messages) { // clear banners and return }`.

Then we don't have most of the code in the function indented more than it needs to be.